### PR TITLE
OpenAPI code generation

### LINKED ISSUE #101: Add `edgespec codegen openapi` command

### PR DESCRIPTION
**Pull request by codetheweb**
This isn't quite at parity with nextlove's generation, I'll make issues for the remaining work after merging.

### CODE REVIEW

File: src/create-with-edge-spec.ts [hunk 1]
```diff
@@ -20,70 +20,76 @@ import { ResponseTypeToContext } from "./types/context.js"
...
+          // validating them against the routeSpec
+          serializeResponse(globalSpec, routeSpec),
+        ],
+        routeFn,
+        request,
+        ctx
+      )
+    }
+
+    requestHandler._routeSpec = routeSpec
```
Comment (mxsdev) [inline src/create-with-edge-spec.ts]: Can we convert this to a functional pattern? i.e. `withAttachRouteSpec(routeSpec, ...)`

I think that would be more readable

File: tests/cli/codegen/openapi/openapi-codegen.test.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  )
+  // ...but not required in the query for POST...
+  t.falsy(
+    (
+      post?.parameters?.find(
+        (p) => (p as ParameterObject).name === "user_id"
+      ) as ParameterObject
+    ).required
+  )
+  // ...and it should be optional in the POST body
```
Comment (mxsdev) [inline tests/cli/codegen/openapi/openapi-codegen.test.ts:76]: test that get body is null?

File: src/cli/commands/codegen/openapi.ts [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    description: "Path to the output file",
+    required: true,
+  })
+
+  async run(config: ResolvedEdgeSpecConfig) {
+    const tempBundlePath = path.join(os.tmpdir(), `${randomUUID()}.js`)
+    await fs.writeFile(tempBundlePath, await bundle(config))
+    const runtimeBundle = await loadBundle(tempBundlePath)
+
+    const { project, globalRouteSpecType } = await extractRouteSpecsFromAST({
```
Comment (mxsdev) [inline src/cli/commands/codegen/openapi.ts]: Switch to just using the bundled routes

File: tests/cli/codegen/openapi/openapi-codegen.test.ts (hunk 2 above)
Comment (codetheweb) [reply tests/cli/codegen/openapi/openapi-codegen.test.ts:76]: if the get body is populated the previous test fails 👍

File: src/create-with-edge-spec.ts (hunk 1 above)
Comment (codetheweb) [reply src/create-with-edge-spec.ts]: yeah that makes sense

Comment (codetheweb) [reply src/create-with-edge-spec.ts]: 484dc88

File: src/cli/commands/codegen/openapi.ts (hunk 3 above)
Comment (codetheweb) [reply src/cli/commands/codegen/openapi.ts]: 9f0b9c4

**Review by mxsdev — APPROVED** [review]

**Pull request merged**